### DISCUSSION TITLE
jupyter notebook walkthrough troubleshooting

### DISCUSSION THREAD

**Discussion by clauvasq** [tbepler/topaz#83] [Q&A]
Hi All,
I'm trying to go through the jupyter notebook '02_walkthrough' after installing topaz using conda. And I'm having some issues running the fifth code cell. I think it has to do with how the package is calling bash...but I'm not sure what's gone wrong, let alone how to fix it (I'm working on a Windows10 computer but I did get bash working -- at least the mkdir cell before this one that I'm stuck on works!).

I've included the errors I get below, I'm pretty sure this is a setup problem. But any insight would be very welcome.

Thanks in advance!
Claudia

---
/etc/profile.d/conda.sh: No such file or directoryine 5: C:/Users/DunnLab/anaconda3
: numeric argument requiredda3/Scripts/activate: line 5: return: 1
/bin/bash: /mnt/c/Users/DunnLab/anaconda3/envs/topaz/Scripts/topaz: C:/Users/DunnLab/anaconda3/envs/topaz/bin/python: bad interpreter: No such file or directory
---------------------------------------------------------------------------
CalledProcessError                        Traceback (most recent call last)
<ipython-input-4-b14928314bed> in <module>
----> 1 get_ipython().run_cell_magic('bash', '', 'source activate topaz\n# conda activate topaz\n\n# To run the preprocess command, we pass the input micrographs as command line arguments.\n# Preprocess will write the processed images to the directory specified with the -o argument.\n\n# Preprocess normalizes micrographs using a 2 component Gaussian mixture model. This model is robust to\n# dark pixels from grid edges during normalization. To use a conventional affine normalization,\n# set the --affine flag.\n\n# -s sets the downsampling amount (in this case, we downsample by 8)\n# -d/--device X sets preprocess to use GPU with ID X\n# -t/--num-workers X sets preprocess to use X threads, this and GPU device are mutually exclusive\n# -v gives verbose output to track progress\n\ntopaz preprocess -s 8 -o data/EMPIAR_10025/processed/micrographs/ data/EMPIAR_10025/rawdata/micrographs/*.mrc\n')

~\anaconda3\envs\topaz\lib\site-packages\IPython\core\interactiveshell.py in run_cell_magic(self, magic_name, line, cell)
   2369             with self.builtin_trap:
   2370                 args = (magic_arg_s, cell)
-> 2371                 result = fn(*args, **kwargs)
   2372             return result
   2373 

~\anaconda3\envs\topaz\lib\site-packages\IPython\core\magics\script.py in named_script_magic(line, cell)
    140             else:
    141                 line = script
--> 142             return self.shebang(line, cell)
    143 
    144         # write a basic docstring:

<decorator-gen-111> in shebang(self, line, cell)

~\anaconda3\envs\topaz\lib\site-packages\IPython\core\magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

~\anaconda3\envs\topaz\lib\site-packages\IPython\core\magics\script.py in shebang(self, line, cell)
    243             sys.stderr.flush()
    244         if args.raise_error and p.returncode!=0:
--> 245             raise CalledProcessError(p.returncode, cell, output=out, stderr=err)
    246 
    247     def _run_script(self, p, cell, to_close):

CalledProcessError: Command 'b'source activate topaz\n# conda activate topaz\n\n# To run the preprocess command, we pass the input micrographs as command line arguments.\n# Preprocess will write the processed images to the directory specified with the -o argument.\n\n# Preprocess normalizes micrographs using a 2 component Gaussian mixture model. This model is robust to\n# dark pixels from grid edges during normalization. To use a conventional affine normalization,\n# set the --affine flag.\n\n# -s sets the downsampling amount (in this case, we downsample by 8)\n# -d/--device X sets preprocess to use GPU with ID X\n# -t/--num-workers X sets preprocess to use X threads, this and GPU device are mutually exclusive\n# -v gives verbose output to track progress\n\ntopaz preprocess -s 8 -o data/EMPIAR_10025/processed/micrographs/ data/EMPIAR_10025/rawdata/micrographs/*.mrc\n'' returned non-zero exit status 126.

**Comment by tbepler** [comment]
Did you install topaz with anaconda?

**Reply by clauvasq** [reply]
yep!

**Reply by tbepler** [reply]
It looks like a problem with jupyter/your python installation, because bash can't find your python installation.

**Reply by tbepler** [reply]
If you try running the commands through the terminal directly, do you get any errors? If you aren't using windows subsystem for linux (WSL), that may make things easier.

**Reply by clauvasq** [reply]
hmmm,yea, it looks like you're right. I get this from running at the command prompt:
`(topaz) C:\Users\DunnLab\Documents\topaz>topaz preprocess -s 8 -o data/EMPIAR-10025/processed/micrographs/ data/EMPIAR-10025/rawdata/micrographs/*.mrc
'topaz' is not recognized as an internal or external command,
operable program or batch file.`

**Comment by JochenKa** [comment]
Hi, I'm running into the exact same problem, also on Windows 10 and installed with anaconda. Did you find a solution?
